SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp

```diff
@@ -567,7 +567,8 @@ int64 USpatialActorChannel::ReplicateActor()
 		NetDriver->LoadBalanceStrategy != nullptr &&
 		// TODO: the 'bWroteSomethingImportant' check causes problems for actors that need to transition in groups (ex. Character, PlayerController, PlayerState),
 		// so disabling it for now.  Figure out a way to deal with this to recover the perf lost by calling ShouldChangeAuthority() frequently. [UNR-2387]
-		Actor->HasAuthority() &&
+		NetDriver->StaticComponentView->HasAuthority(EntityId, SpatialConstants::AUTHORITY_INTENT_COMPONENT_ID) &&
+		!NetDriver->LockingPolicy->IsLocked(Actor) &&
```

Comment:
Nit: should there be `NetDriver->LockingPolicy != nullptr` check?

Reply:
If something doesn't already guarantee that can't be the case.

Reply:
`SpatialGDKSettings->bEnableUnrealLoadBalancer` is the gate for whether the `LockingPolicy` is created

Reply:
Should `NetDriver->LoadBalanceStrategy != nullptr` be removed then? :)

Reply:
Makes sense to me